### PR TITLE
GEODE-5971: Refactor AlterAsyncEventQueueCommand to extend SingleGfshCommand

### PR DESCRIPTION
**Pull request by pdxrunner**
The following changes were required to implement the configuration updates for a command that does not allow specifying a `--group` option.
- Rename `SingleGfshCommand.updateClusterConfig` to `updateGroupConfig`, and change signature to return `boolean` instead of `void`.
- Add method `SingleGfshCommand.updateAllConfigs` to support configuration updates by commands that don't recognize groups.

This PR includes the proposed changes from https://github.com/apache/geode/pull/2825, with the exception of adding a `ResultModel` argument to `updateClusterConfig`, which was found to not be needed.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java [hunk 1]
```diff
@@ -152,6 +161,17 @@ protected Object invokeCommand(Object command, GfshParseResult parseResult) {
         return cc;
       });
     }
+
+    Map<String, CacheConfig> configMap = ccService.getGroups()
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java]: I don't quite see how this code is reachable. If we have `groups` passed in, then the above block is run. If no groups are set then the group is defaulted to `cluster` and the above block is run. Maybe I'm missing something?

Comment (pdxrunner) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java]: After all the groups specified (or "cluster") are cycled through in the `for` loop we drop into this code. The `*Command` implementor can override either `updateGroupConfig` or `updateAllConfigs`. If `updateGroupConfigs` is not overridden, then nothing gets processed in the `for` loop and everything gets processed via this `Map`. See the `AlterAsyncEventQueueCommand` changes for an `updateAllConfigs` implementation.

Comment (jdeppe-pivotal) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/remote/CommandExecutor.java]: Doh - I missed that the return was inside the closure and not inside the for loop. NM.

**Comment by pdxrunner** [comment]
Closing the PR to rework tests that are directly manipulating configuration XML

**Pull request closed without merging**